### PR TITLE
feat(storage): support huaweicloud obs as storage

### LINKED ISSUE #7245: feat: support huaweicloud obs as storage
**Summary**

https://github.com/datafuselabs/opendal/pull/557 merged, it's time to make databend supporting obs as storage.

### PR DESCRIPTION
**Pull request by eastfisher**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support huaweicloud obs as storage.

Fixes #7245

### CODE REVIEW

**Review by ClSlaid — APPROVED** [review]
Nice and complete.

**Comment by Xuanwo** [comment]
This PR depends on my ongoing work to bump opendal which will be proposed tomorrow.

**Comment by Xuanwo** [comment]
@eastfisher Hi, sorry for the waiting, we can restart now!

File: src/query/config/src/outer_v0.rs [hunk 1]
```diff
@@ -573,6 +584,85 @@ impl TryInto<InnerStorageHdfsConfig> for HdfsConfig {
...
+    pub bucket: String,
+
+    /// Endpoint URL for OBS
+    ///
+    /// # TODO(xuanwo)
+    ///
+    /// Clap doesn't allow us to use endpoint_url directly.
+    #[clap(long = "storage-obs-endpoint-url", default_value_t)]
+    #[serde(rename = "endpoint_url")]
+    pub endpoint_url: String,
```
Comment (Xuanwo) [inline src/query/config/src/outer_v0.rs:609]: Due to the limitation of `clap`, we can't use the same field with other structs. Please add an `obs_` prefix for them.

Comment (eastfisher) [reply src/query/config/src/outer_v0.rs]: fixed

File: docs/doc/50-manage/01-query/10-query-config.md [hunk 2]
```diff
@@ -191,6 +191,36 @@ If `endpoints` is configured, the `address` configuration will no longer take ef
 * Env variable: `STORAGE_AZBLOB_ACCOUNT_KEY`
 * Required.
 
+### storage.obs
+
+#### bucket
+
+* OBS bucket name.
+* Default: `""`
+* Env variable: `STORAGE_OBS_BUCKET`
```
[the remark this answers is not in the document]
Comment (eastfisher) [reply docs/doc/50-manage/01-query/10-query-config.md:200]: OK, thanks!

**Review by Xuanwo — APPROVED** [review]
Mostly LGTM

**Comment by Xuanwo** [comment]
@eastfisher Congratulation!

**Comment by Xuanwo** [comment]
Cc @soyeric128, are you willing to add a deploy chapter for obs support?

**Comment by soyeric128** [comment]
> Cc @soyeric128, are you willing to add a deploy chapter for obs support?

Hi @Xuanwo Sorry, I saw this PR already updated the document. Something else needs to be documented?

**Pull request merged**